include/c3d/effect.h

```diff
@@ -1,15 +1,80 @@
+ * @param[in] pass  Specifies the stencil action when both the stencil test and the depth test pass, or when the stencil test passes and either there is no depth buffer or depth testing is not enabled. The initial value is \ref GPU_STENCIL_KEEP.
+ */
 void C3D_StencilOp(GPU_STENCILOP sfail, GPU_STENCILOP dfail, GPU_STENCILOP pass);
+
+/**
+ * @brief Set the blend color
+ * @param[in] color Specifies the RGBA blend color. The initial value is 0.
+ */
 void C3D_BlendingColor(u32 color);
+
 void C3D_EarlyDepthTest(bool enable, GPU_EARLYDEPTHFUNC function, u32 ref);
+
+/**
+ * Configure depth testing options.
+ * @note Setting the enable parameter to false will not also disable depth writes. It will instead behave as if the depth function were set to \ref GPU_ALWAYS. To completely disable depth-related operations, the enable parameter must be false, and the writemask should be \ref GPU_WRITE_COLOR.
+ * @param[in] enable    If true, do depth comparisons on the outgoing fragments and write to the depth buffer. The initial value is true.
+ * @param[in] function  Specifies the depth comparison function. The initial value is \ref GPU_GREATER.
+ * @param[in] writemask Configures buffer writemasks for the depth test stage. The initial value is \ref GPU_WRITE_ALL.
+ */
 void C3D_DepthTest(bool enable, GPU_TESTFUNC function, GPU_WRITEMASK writemask);
+
+/**
+ * @brief Configure Alpha testing
+ * @param[in] enable   Enables or disables alpha test. The initial value is false.
+ * @param[in] function Specifies the alpha comparison function. The initial value is \ref GPU_ALWAYS.
+ * @param[in] ref      Specifies the reference value that incoming alpha values are compared to (0 disables depth buffer writing). The intial value is 0.
+ */
 void C3D_AlphaTest(bool enable, GPU_TESTFUNC function, int ref);
+
+/**
+ * @brief Configure blend functions
+ * @param[in] colorEq  Specifies how source and destination colors are combined. The initial value is \ref GPU_BLEND_ADD.
+ * @param[in] alphaEq  Specifies how source and destination alphas are combined. The initial value is \ref GPU_BLEND_ADD.
+ * @param[in] srcClr   Specifies how the red, green, and blue source blending factors are computed. The initial value is \ref GPU_SRC_ALPHA.
+ * @param[in] dstClr   Specifies how the red, green, and blue destination blending factors are computed. The initial value is \ref GPU_ONE_MINUS_SRC_ALPHA.
+ * @param[in] srcAlpha Specifies how the alpha source blending factors are computed. The initial value is \ref GPU_SRC_ALPHA.
+ * @param[in] dstAlpha Specifies how the alpha destination blending factors are computed. The initial value is \ref GPU_ONE_MINUS_SRC_ALPHA.
+ */
 void C3D_AlphaBlend(GPU_BLENDEQUATION colorEq, GPU_BLENDEQUATION alphaEq, GPU_BLENDFACTOR srcClr, GPU_BLENDFACTOR dstClr, GPU_BLENDFACTOR srcAlpha, GPU_BLENDFACTOR dstAlpha);
+
 void C3D_ColorLogicOp(GPU_LOGICOP op);
+
 void C3D_FragOpMode(GPU_FRAGOPMODE mode);
+
```

Comment:
couple of other missing docs here as well, it looks like, but these functions also seem to be pretty self-explanatory so it may not be worth adding.